tests/store/artifact/test_azure_data_lake_artifact_repo.py

```diff
@@ -147,18 +180,74 @@ def test_log_artifacts(mock_filesystem_client, mock_directory_client, tmp_path):
+        posixpath.join(TEST_ROOT_PATH, "."),
+        posixpath.join(TEST_ROOT_PATH, "subdir"),
+        posixpath.join(TEST_ROOT_PATH, "subdir"),
+    ]
+
+    uploaded_filenames = [
+        call[0][0] for call in mock_directory_client.get_file_client.call_args_list
+    ]
+    assert len(uploaded_filenames) == 3
+    assert set(uploaded_filenames) == {"a.txt", "b.txt", "empty-file.txt"}
+
+    mock_directory_client.get_file_client("a.txt").upload_data.assert_called()
+    mock_directory_client.get_file_client("b.txt").upload_data.assert_called()
+    mock_directory_client.get_file_client("subdir/empty-file.txt").create_file.assert_called()
 
-    # Ensure that we uploaded all the expected files
-    uploaded_filenames = [call[0][0] for call in call_list]
-    nonempty_files = {"./a.txt", "subdir/b.txt"}
-    empty_files = {"subdir/empty-file.txt"}
-    assert set(uploaded_filenames) == nonempty_files.union(empty_files)
 
-    mock_directory_client.get_file_client("./a.txt").upload_data.assert_called()
-    mock_directory_client.get_file_client("subdir/b.txt").upload_data.assert_called()
-    mock_directory_client.get_file_client("subdir/empty-file.txt").create_file.assert_called()
+def test_log_artifacts_in_parallel_when_necessary(tmp_path):
+    fake_sas_token = "fake_session_token"
+    repo = AzureDataLakeArtifactRepository(TEST_DATA_LAKE_URI, AzureSasCredential(fake_sas_token))
+
+    parentd = tmp_path.joinpath("data")
+    parentd.mkdir()
+    parentd.joinpath("a.txt").write_text("ABCDE")
+
+    with mock.patch(f"{ADLS_REPOSITORY_PACKAGE}._MULTIPART_UPLOAD_CHUNK_SIZE", 0), mock.patch(
+        f"{ADLS_ARTIFACT_REPOSITORY}._multipart_upload", return_value=None
+    ) as multipart_upload_mock, mock.patch(
+        f"{ADLS_ARTIFACT_REPOSITORY}.log_artifact", return_value=None
+    ):
+        repo.log_artifacts(parentd)
+        multipart_upload_mock.assert_called_with(
+            ArtifactCredentialInfo(
+                signed_uri="https://account.dfs.core.windows.net/filesystem/some/path/"
+                + "./a.txt?fake_session_token"
+            ),
+            ANY,
+            "./a.txt",
+        )
+
+
+@pytest.mark.parametrize(
+    ("file_size", "is_parallel_download"),
+    [(None, False), (100, False), (499_999_999, False), (500_000_000, True)],
+)
+def test_download_file_in_parallel_when_necessary(file_size, is_parallel_download):
+    repo = AzureDataLakeArtifactRepository(TEST_DATA_LAKE_URI, None)
+    remote_file_path = "file_1.txt"
+    list_artifacts_result = (
+        [FileInfo(path=remote_file_path, is_dir=False, file_size=file_size)] if file_size else []
+    )
+    with mock.patch(
+        f"{ADLS_ARTIFACT_REPOSITORY}.list_artifacts",
+        return_value=list_artifacts_result,
+    ), mock.patch(
+        f"{ADLS_ARTIFACT_REPOSITORY}._download_from_cloud", return_value=None
+    ) as download_mock, mock.patch(
+        f"{ADLS_ARTIFACT_REPOSITORY}._parallelized_download_from_cloud", return_value=None
```

Comment:
We do not test the internals of `_parallelized_download_from_cloud` following the status quo in `test_databricks_artifact_repo.py`.